src/pixie_cli/pkg/cmd/root.go

```diff
@@ -185,6 +198,42 @@ var RootCmd = &cobra.Command{
 	},
 }
 
+// Name a variable to store a slice of commands that don't require cloudAddr
+var cmdsCloudAddrNotReqd = []*cobra.Command{
+	CollectLogsCmd,
+	VersionCmd,
+}
+
+func getCloudAddrIfRequired(cmd *cobra.Command) string {
+	// Commands within allow list should be opted out in addition to Cobra's
+	// default help command
+	if slices.Contains(cmdsCloudAddrNotReqd, cmd) || cmd.Short == "Help about any command" {
+		return defaultCloudAddr
+	}
+
+	cloudAddr := viper.GetString("cloud_addr")
+	if cloudAddr == "" {
+		if !isatty.IsTerminal(os.Stdin.Fd()) {
+			utils.Errorf("No cloud address provided during run within interactive shell. Please set the cloud address using the `--cloud_addr` flag or `PX_CLOUD_ADDR` environment variable.")
```

Comment:
error says `interactive shell` but this isn't a tty right?

Reply:
Good call, I meant to say non-interactive shell.